src/snippets/nosecone/quick-start/next-js/Step2.mdx

```diff
@@ -0,0 +1,65 @@
+import { Code } from "@astrojs/starlight/components";
+import SelectableContent from "@/components/SelectableContent";
+
+import Step2MiddlewareTS from "./Step2Middleware.ts?raw";
+import Step2MiddlewareJS from "./Step2Middleware.js?raw";
+
+import Step2LayoutConnectionTS from "./Step2LayoutConnection.ts?raw";
+import Step2LayoutConnectionJS from "./Step2LayoutConnection.js?raw";
+import Step2LayoutNoCacheTS from "./Step2LayoutNoCache.ts?raw";
+import Step2LayoutNoCacheJS from "./Step2LayoutNoCache.js?raw";
+
+Nosecone applies headers to all your routes via middleware in your Next.js
+application.
+
+<SelectableContent client:load syncKey="language" frameworkSwitcher>
+  <div slot="TS" slotIdx="1">
+    <Code code={Step2MiddlewareTS} lang="ts" title="middleware.ts" />
+  </div>
+  <div slot="JS" slotIdx="2">
+    <Code code={Step2MiddlewareJS} lang="js" title="middleware.js" />
+  </div>
+</SelectableContent>
+
+:::caution
+We recommend you remove any `export const config = ...` from your middleware so
+Nosecone will run on every route.
+:::
+
+You also need to opt-out of static generation in your application. To do this,
```

Comment:
Worth explaining why this is and the tradeoffs.

Reply:
I thought we avoided that level of details in getting started. I was planning to describe in the nextjs adapter reference docs.

Reply:
Yeah that's fine - we can include a link the explanation.

Reply:
I added a link to the reference guide.